### PR TITLE
Add support for google pay

### LINKED ISSUE #58: Add support for Google Pay
The payment flow should work analogous to our Apple Pay flow.
 
Documentation can be found [here](https://stripe.com/docs/google-pay).

### PR DESCRIPTION
**Pull request by felixbrucker**
This PR adds support for google pay. As the flow is pretty much the same as apple pay i decided to merge them into one handler (`digital wallet payments`) and also update the apple pay payments to use payment intents.

Closes #58

### CODE REVIEW

File: Components/PaymentMethods/DigitalWalletPayments.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                'metadata' => [
+                    'platform_name' => Util::STRIPE_PLATFORM_NAME,
+                ],
+            ]);
+        }
+        $user = Shopware()->Session()->sOrderVariables['sUserData'];
+        $userEmail = $user['additional']['user']['email'];
+        $customerNumber = $user['additional']['user']['customernumber'];
+
+        // Use the token to create a new Stripe card payment intent
```
Comment (JanRogge) [inline Components/PaymentMethods/DigitalWalletPayments.php]: There seems to be no token. Do you mean payment_method?

File: Views/frontend/checkout/confirm.tpl [hunk 2]
```diff
@@ -43,39 +43,56 @@
...
+                        stripePaymentSnippets.error.connectionNotSecure = stripePaymentSnippets.error.connectionNotSecure.replace('[0]', 'Apple Pay');
+                        stripePaymentSnippets.error.invalidConfig = stripePaymentSnippets.error.invalidConfig.replace('[0]', 'Apple Pay');
+                        stripePaymentSnippets.error.notAvailable = stripePaymentSnippets.error.notAvailable.replace('[0]', 'Apple Pay');
+                        break;
+                    case 'google_pay':
+                        stripePaymentSnippets.error.connectionNotSecure = stripePaymentSnippets.error.connectionNotSecure.replace('[0]', 'Google Pay');
+                        stripePaymentSnippets.error.invalidConfig = stripePaymentSnippets.error.invalidConfig.replace('[0]', 'Google Pay');
+                        stripePaymentSnippets.error.notAvailable = stripePaymentSnippets.error.notAvailable.replace('[0]', 'Google Pay');
+                        break;
+                }
```
Comment (svenmuennich) [inline Views/frontend/checkout/confirm.tpl]: How about deciding on the payment method title first and then replace it in the string only once?

File: Views/frontend/_public/src/javascript/stripe_payment_digital_wallet_payments.js [hunk 3]
```diff
@@ -93,43 +119,55 @@ var StripePaymentApplePay = {
...
 
-        // Check for availability of Apple Pay
+        // Check for availability of the payment api
         me.paymentRequest.canMakePayment().then(function (result) {
-            me.applePayAvailable = result && result.applePay;
-            if (!me.applePayAvailable) {
+            if (me.selectedPaymentMethod === 'apple_pay') {
+                me.paymentApiAvailable = result && result.applePay;
+            } else {
+                me.paymentApiAvailable = !!result && !result.applePay;
```
[the remark this answers is not in the document]
Comment (felixbrucker) [reply Views/frontend/_public/src/javascript/stripe_payment_digital_wallet_payments.js]: As `result` will be null or an object we dont need it here specifically. If we do not care about users paying with apple pay with the google pay payment method selected we can also remove the `applePay` false check.

File: Components/PaymentMethods/DigitalWalletPayments.php [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+        // Enable receipt emails, if configured
+        $pluginConfig = $this->get('plugins')->get('Frontend')->get('StripePayment')->Config();
+        if ($pluginConfig->get('sendStripeChargeEmails')) {
+            $paymentIntentConfig['receipt_email'] = $userEmail;
+        }
+
+        $paymentIntent = Stripe\PaymentIntent::create($paymentIntentConfig);
+        if (!$paymentIntent) {
+            throw new \Exception($this->getSnippet('payment_error/message/transaction_not_found'));
```
[the remark this answers is not in the document]
Comment (felixbrucker) [reply Components/PaymentMethods/DigitalWalletPayments.php]: Actually we dont even need this, i just removed it

**Pull request merged**